client/src/game/reducers/getSoundType.ts

```diff
@@ -90,28 +90,24 @@ export default function getSoundType(
         return SoundType.DiscardClued;
       }
 
-      const { lastAction } = originalState.stats;
-      let couldBeLastDiscardedCard = true;
-      if (
-        lastAction !== null &&
-        lastAction.type === "discard" &&
-        lastAction.suitIndex !== null &&
-        lastAction.rank !== null
-      ) {
-        couldBeLastDiscardedCard = discardedCard.possibleCardsFromClues.some(
-          ([suitIndex, rank]) =>
-            suitIndex === lastAction.suitIndex && rank === lastAction.rank,
-        );
-      }
       const nextPlayerHand = currentState.hands[action.playerIndex];
       if (
         originalState.stats.doubleDiscard !== null &&
-        couldBeLastDiscardedCard &&
         !metadata.hardVariant &&
         !handRules.isLocked(nextPlayerHand, currentState.deck)
       ) {
-        // A player has discarded *in* a double discard situation
-        return SoundType.DoubleDiscard;
+        const previouslyDiscardedCard =
+          originalState.deck[originalState.stats.doubleDiscard];
+        if (
+          cardRules.canPossiblyBe(
```

Comment:
Does canPossiblyBe use asymmetric information? I think it might have been intentional that the old code only considered explicit information from clues.

Reply:
```ts
export function canPossiblyBe(
  card: CardState,
  suitIndex: number | null,
  rank: number | null,
): boolean {
  if (suitIndex === null && rank === null) {
    // We have nothing to check
    return true;
  }
  return card.possibleCardsForEmpathy.some(
    ([s, r]) =>
      (suitIndex === null || suitIndex === s) && (rank === null || rank === r),
  );
}
```

Reply:
its using empathy, thus it appears that it only uses symmetric information.

the fact that timo had to ask this question means that the function is incorrectly named, which is a bug.
can you rename it to something better?
maybe `canPossiblyBeFromEmpathy()`

Reply:
`canPossiblyBe` a wrapper around `possiblyCardsForEmpathy`, so it's whatever the empathy shows.

We can change it to use `possibleCardsFromClues` instead but then I'd need to tweak the question mark code as well.

Reply:
```ts
  // possibleCardsForEmpathy is a one-dimensional array of tuples
  // It contains a tuple for each specific card that is still possible based on everything the
  // player holding it should know so far
  // Do not access this by the index; filter the array to find the remaining cards that you need
  readonly possibleCardsForEmpathy: ReadonlyArray<readonly [number, number]>;
```

ok so its based on assync info, thats bad

Reply:
> We can change it to use possibleCardsFromClues instead but then I'd need to tweak the question mark code as well.

Yes, as I've said earlier, both the sounds and the question mark code should not use assync info

Reply:
huh. didn't realize the empathy used async info, ok